flipdisc/framework/app.py

```diff
@@ -358,134 +400,126 @@ async def notify(self, channel, data):
         if not self._red:
             raise Exception("redis not enabled")
         self._log.debug("Redis outgoing, channel: %s, data: %s", channel, data)
-        return await self._red.publish(channel, json.dumps(data))
+        return await self._red.pubsub.publish(channel, json.dumps(data))
 
-    def set_redis_callback(self, function):
+    def set_redis_callback(self, function: Callable):
         """
         Define a callback to be invoked on messages received through
         the redis pubsub.
         """
         self._redis_sub_callback = function
 
-    def add_periodic_callback(self, function, float_sec, start=False):
+    def add_periodic_callback(self, function: Callable, float_sec: float, start: bool=False) -> int:
         """
         Schedule function to be called once each n seconds. The callback
         will receive this instance as its sole argument.
 
         The returned key can be used with stop_periodic_callback to stop
         the scheduling.
         """
-        cb = ioloop.PeriodicCallback(
-            lambda: function(self), float_sec * 1000
-        )  # convert to ms
-        key = id(cb)
-        self._periodic_callbacks[key] = cb
+        periodic = ScheduledFunction(function, float_sec, True, self._loop, self)
```

Comment:
See below note on the ScheduledFunction constructor for context here :)

```suggestion
        periodic = ScheduledFunction(function, [self], float_sec, periodic=True, loop=self._loop)
```